notebooks/03-query/11-joins.ipynb

```diff
@@ -0,0 +1,2475 @@
+    "lookup_definition_version = datetime.now().strftime(\"%Y-%m-%dT%H:%M:%SZ\")\n",
+    "lookup_definition = {\n",
+    "    lookup_tier: {\n",
+    "        lookup_name: {\n",
+    "            \"version\": lookup_definition_version,  \n",
+    "            \"lookupExtractorFactory\": {\n",
+    "                \"type\": \"map\",\n",
+    "                    \"map\": map\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    "}\n",
+    "postLookup(lookup_definition)\n",
+    "waitForLookup(lookup_tier, lookup_name, 30)\n",
+    "print_timestamp()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6bf7f21b-7cdd-460e-bc00-e74037e21717",
+   "metadata": {},
+   "source": [
+    "## Prep Query Testing and Measurement Function\n",
+    "We measure each method of joining the data by running the query multiple times and avoiding cacheing of the results in order to get a avg, min and max runtimes of each query. We'll use the following function to achieve this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "38f8b68a-52ca-41bf-9d84-ea668632528c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from statistics import mean \n",
+    "\n",
+    "def measure_query( sql: str, iterations: int ):\n",
+    "    req = sql_client.sql_request(sql)\n",
+    "    req.add_context(\"populateCache\", \"false\")  # run without cacheing results to get a real sense of performance\n",
+    "    req.add_context(\"useCache\", \"false\")  # do not use cached results\n",
+    "    stats = []\n",
+    "    while (iterations>0):\n",
+    "      start = datetime.now()\n",
+    "      sql_client.sql(req)\n",
+    "      end = datetime.now()\n",
+    "      stats.append( (end - start).total_seconds() * 10**3 ) # add run time in milliseconds\n",
+    "      iterations -=1\n",
+    "    return f\"Results = avg:{mean(stats)} ms   min:{min(stats)} ms  max:{max(stats)} ms\""
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "4cf7592d-b852-4712-b8ec-60be9df1224d",
+   "metadata": {},
+   "source": [
+    "## Synchronous Queries on Scatter/Gather Engine\n",
+    "Druid's scatter/gather query processing is designed for speed.\n",
+    "### The Scatter/Gather Engine\n",
+    "At a high level, Druid queries run as follows:\n",
+    "\n",
+    "<img src=\"assets/scatter-gather.png\" alt=\"drawing\" style=\"width:600px;\"/>\n",
+    "\n",
+    "1. Broker receives query\n",
+    "2. Broker plans query and determines which segments are relevant using cached segment metadata\n",
+    "3. Broker forwards query to data servers, telling each server what segments to interrogate\n",
+    "4. Data servers process segment files\n",
+    "5. Data servers send results back to Broker\n",
+    "6. Broker performs final processing and returns results\n",
+    "\n",
+    "Note that the data servers do not directly communicate with each other.\n",
+    "\n",
+    "### Joins with a Scatter/Gather Engine\n",
```

Comment:
I think this should be "Joins with the synchronous query API"? @techdocsmith?

You could go with something like:

The cell below contains an XXXX-type JOIN. It has:

* Chips for tea
* Eggs for breakfast
* A INNER JOIN

Execute it on the synchronous query API by running the next cell.

Reply:
Have we agreed on "Asynchronous/Synchronous Queries" vs "Native vs MSQ"? I tried for the middle ground Scatter/Gather vs Multi-Stage... glad to change, but I do think we need a standard.